### DISCUSSION TITLE
Remove the 2Mb wasm size limit for local deployments

### DISCUSSION THREAD

**Discussion by 0kenx** [dfinity/sdk#3174] [Feature Requests]
Is it possible to get away with the 2Mb wasm size limit for local deployments? It's time consuming to run ic-wasm when I test locally, additionally when deploying on local I want to include extra debug-only code that would make the code size bigger than release. It would help to remove the 2Mb wasm size limit for local deployments and replace with a warning instead.

**Comment by viviveevee** [comment]
I'm not sure if it helps, but if you run the replica in system subnet mode you may get a higher limit. Aside: work is ongoing to allow larger (I heard something about tens of MB at some point) wasms even on mainnet

**Comment by ghost** [comment]
Implemented in https://github.com/dfinity/sdk/pull/3511 and available in dfx 0.16.0
